components/eam/src/chemistry/mozart/mo_drydep.F90

```diff
@@ -2523,702 +975,682 @@ subroutine drydep_xactive( ncdate, sfc_temp, pressure_sfc,  &
+
+    do ispec = 1,gas_pcnst
        if( has_dvel(ispec) ) then
-          m = map_dvel(ispec)
+          idx_drydep = map_dvel(ispec)
           do lt = beglt,endlt
-             do i = 1,ncol
-                if( fr_lnduse(i,lt) ) then
-                   sndx = index_season(i,lt)
+             do icol = 1,ncol
+                if( fr_lnduse(icol,lt) ) then
+                   sndx = index_season(icol,lt)
                    if( ispec == o3_ndx .or. ispec == o3a_ndx .or. ispec == so2_ndx ) then
                       rmx = 0._r8
                    else
-                      rmx = 1._r8/(heff(i,m)/3000._r8 + 100._r8*foxd(m))
-                   end if
-                   cts(i) = 1000._r8*exp( - tc(i) - 4._r8 )                 ! correction for frost
-                   rgsx(i,lt,ispec) = cts(i) + 1._r8/((heff(i,m)/(1.e5_r8*rgss(sndx,lt))) + (foxd(m)/rgso(sndx,lt)))
-                   !-------------------------------------------------------------------------------------
-                   ! special case for H2 and CO;; CH4 is set ot a fraction of dv(H2)
-                   !-------------------------------------------------------------------------------------
-                   if( ispec == h2_ndx .or. ispec == co_ndx .or. ispec == ch4_ndx ) then
-                      if( ispec == co_ndx ) then
-                         fact_h2 = 1.0_r8
-                      elseif ( ispec == h2_ndx ) then
-                         fact_h2 = 0.5_r8
-                      elseif ( ispec == ch4_ndx ) then
-                         fact_h2 = 50.0_r8
-                      end if
-                      !-------------------------------------------------------------------------------------
-                      ! no deposition on snow, ice, desert, and water
-                      !-------------------------------------------------------------------------------------
-                      if( lt == 1 .or. lt == 7 .or. lt == 8 .or. sndx == 4 ) then
-                         rgsx(i,lt,ispec) = large_value
-                      else
-                         var_soilw = max( .1_r8,min( soilw(i),.3_r8 ) )
-                         if( lt == 3 ) then
-                            var_soilw = log( var_soilw )
-                         end if
-                         dv_soil_h2 = h2_c(lt) + var_soilw*(h2_b(lt) + var_soilw*h2_a(lt))
-                         if( dv_soil_h2 > 0._r8 ) then
-                            rgsx(i,lt,ispec) = fact_h2/(dv_soil_h2*1.e-4_r8)
-                         end if
-                      end if
-                   end if
+                      rmx = 1._r8/(heff(icol,idx_drydep)/3000._r8 + 100._r8*foxd(idx_drydep))
+                   endif
+                   cts(icol) = 1000._r8*exp( - tc(icol) - 4._r8 )                 ! correction for frost
+                   rgsx(icol,lt,ispec) = cts(icol) + 1._r8/((heff(icol,idx_drydep)/(1.e5_r8*rgss(sndx,lt))) + (foxd(idx_drydep)/rgso(sndx,lt)))
+
                    if( lt == 7 ) then
```

Comment:
Is `lt==7` something special, if it is, we can declare `7` as a named constant (a parameter at the top)

Reply:
make it parameter `lt_for_water`